modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BinarySpatialOpTypeBinding.java

```diff
@@ -55,6 +62,18 @@ public Class getType() {
     
     @Override
     public Object getProperty(Object object, QName name) throws Exception {
-        return FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        Expression e = FESParseEncodeUtil.getProperty((BinarySpatialOperator) object, name);
+        if (e instanceof Literal && ((Literal) e).getValue() instanceof Geometry) {
+            return new EncoderDelegate() {
+                @Override
+                public void encode(ContentHandler output) throws Exception {                    
+                    Encoder encoder = new Encoder(new org.geotools.gml3.v3_2.GMLConfiguration());
```

Comment:
In other words, the explicit mention of "org.geotools.gml3.v3_2.GML.AbstractGeometry" cannot be avoided, because it is the ntire reason of this block of code. 
This is exactly necessary, because the Encoder has no way to just "know" from the scheme that a geometry must be encoded here (unlike what was case with the old filter scheme) and that why it was previously failing to do so.

So in summary, the lack of dependency of a particular GML version that makes you criticise this block of code, is the exact reason that this block of code is necessary in the first place... 

So, rather you must question the whole api and implementation of the geotools Encoder as it is... which is a whole other can of worms.